clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy

```diff
@@ -346,6 +347,34 @@ abstract class OpenstackClientProvider {
     }
   }
 
+/**
+ * Get a network from the network id.
+ * @param region
+ * @param networkId
+ * @return
+ */
+  Network getNetwork(final String region, final String networkId) {
+    handleRequest {
+      getRegionClient(region).networking().network().list().find { it.id == networkId }
+    }
+  }
+
+  /**
+   * Get an unallocated IP from the network, or if none are found, try to create a new floating IP in the network.
+   * @param region
+   * @param networkName
+   * @return
+   */
+  FloatingIP getOrCreateFloatingIp(final String region, final String networkName) {
+    handleRequest {
+      FloatingIP ip = getRegionClient(region).compute().floatingIps().list().find { !it.fixedIpAddress }
+      if (!ip) {
+        ip = client.useRegion(region).compute().floatingIps().allocateIP(networkName)
```

Comment:
IP can be null from the first call. The second call shouldn't return null if it allocates a new IP without throwing an exception. I can add a check though just to be sure.